conans/client/command.py

```diff
@@ -1017,8 +1017,15 @@ def search(self, *args):
         parser.add_argument("-rev", "--revisions", default=False, action='store_true',
                             help='Get a list of revisions for a reference or a '
                                  'package reference.')
+
         args = parser.parse_args(*args)
 
+        if args.revisions and not self._cache.config.revisions_enabled:
+            raise ConanException("This client doesn't support revisions. Enable this feature setting"
```

Comment:
The client supports revisions, but they are not activated. I would change the message to: 
```
The client doesn't have the revisions feature enabled. Enable this feature ...
```
I also prefer the value to `1` instead of `True`

Reply:
I think this PR could also contain the fix for https://github.com/conan-io/conan/issues/4607
The message probably should be shared. Could make sense to create an exception `RevisionsDisabledException` and catch the error printing the message?

Reply:
I'm not sure if the custom exception would work out of the box, it will inherit from `ConanException`, and in #4607 it will be raised from the `RestApiClient::_get_api` function, I think there is a high chance for that exception to be captured before reaching the `conan_api` world.

So, although it is a good idea (👍 ), I will eventually add it in the other issue